### PR TITLE
Transition to torchvision class for torch object detection examples

### PR DESCRIPTION
**Pull request by Meshford**
### Changes
Changed README.md in examples/torch/object_detection with correct path to data;
There was a transition torchvision class in voc0712.py;
Due to this rewritten Transforms, specified type hints.

### Reason for changes

 Incorrect README.md; 
 Incorrect work was performed with the VOC dataset in object detection task.

### Related tickets

77271

### Tests

Metrics were checked for test and train build

### CODE REVIEW

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

File: examples/torch/object_detection/README.md [hunk 1]
```diff
@@ -21,13 +23,14 @@ pip install -r examples/torch/requirements.txt
 This scenario demonstrates quantization with fine-tuning of SSD300 on VOC dataset.
 
 #### Dataset preparation
-- Download and extract VOC2007 and VOC2012 train/val and test data + devkit from [here](https://pjreddie.com/projects/pascal-voc-dataset-mirror/)
+- Download and extract in one folder VOC2007 and VOC2012 train/val and test data from [here](https://pjreddie.com/projects/pascal-voc-dataset-mirror/)
```
Comment (vshampor) [inline examples/torch/object_detection/README.md:26]: This is too vague. Add an text-based illustration of how the directory structure should look like and which path in this structure corresponds to <path_to_dataset>

File: examples/torch/object_detection/dataset.py [hunk 2]
```diff
@@ -138,11 +137,21 @@ def base_transform(image, size, mean, std, normalize_coef):
 
 
 class BaseTransform:
-    def __init__(self, size, mean, std, normalize_coef):
+    def __init__(self,
+                 size: int,
+                 mean: int,
```
Comment (vshampor) [inline examples/torch/object_detection/dataset.py:142]: I doubt that mean, std and normalize_coef values are expected to be integer only.

File: examples/torch/object_detection/dataset.py [hunk 3]
```diff
@@ -138,11 +137,21 @@ def base_transform(image, size, mean, std, normalize_coef):
...
+                 normalize_coef: int
+                 ):
         self.size = size
         self.mean = np.array(mean, dtype=np.float32)
         self.std = np.array(std, dtype=np.float32)
         self.normalize_coef = normalize_coef
 
-    def __call__(self, image, boxes=None, labels=None):
-        return base_transform(image, self.size, self.mean, self.std, self.normalize_coef), boxes, labels
+    def __call__(self, image: Image, target: list):
```
Comment (vshampor) [inline examples/torch/object_detection/dataset.py:151]: Use `List` as a type hint. Return type hints missing.

File: examples/torch/object_detection/datasets/voc0712.py [hunk 4]
```diff
@@ -38,51 +43,64 @@
...
+    def __init__(self,
+                 class_to_ind: Optional[Callable] = None,
+                 keep_difficult: bool = False
+                 ):
         self.class_to_ind = class_to_ind or dict(
             zip(VOC_CLASSES, range(len(VOC_CLASSES))))
         self.keep_difficult = keep_difficult
 
-    def __call__(self, target, width, height):
+    def __call__(self, image: Image, target: dict, pull: bool = False):
```
Comment (vshampor) [inline examples/torch/object_detection/datasets/voc0712.py:65]: Same

File: examples/torch/object_detection/datasets/voc0712.py [hunk 5]
```diff
@@ -92,85 +110,74 @@ class VOCDetection(data.Dataset):
...
+        return_image_info (bool): sign indicating whether to return the height and width of the image
     """
     classes = VOC_CLASSES
     name = 'voc'
 
-    def __init__(self, root, image_sets=(('2007', 'trainval'), ('2012', 'trainval')), transform=None,
-                 target_transform=VOCAnnotationTransform(keep_difficult=False), return_image_info=False, rgb=True):
+    def __init__(self,
+                 root: str,
+                 image_sets: tuple = (('2007', 'trainval'), ('2012', 'trainval')),
```
Comment (vshampor) [inline examples/torch/object_detection/datasets/voc0712.py:126]: Use `Tuple`

File: examples/torch/object_detection/utils/augmentations.py [hunk 6]
```diff
@@ -51,19 +51,19 @@ class Compose:
...
+            transforms.CenterCrop(10),
+            transforms.ToTensor(),
+            ])
     """
 
     def __init__(self, transforms):
         self.transforms = transforms
 
-    def __call__(self, img, boxes=None, labels=None):
+    def __call__(self, img, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:63]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 7]
```diff
@@ -73,13 +73,13 @@ def __init__(self, lambd):
         assert isinstance(lambd, types.LambdaType)
         self.lambd = lambd
 
-    def __call__(self, img, boxes=None, labels=None):
-        return self.lambd(img, boxes, labels)
+    def __call__(self, img, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:76]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 8]
```diff
@@ -73,13 +73,13 @@ def __init__(self, lambd):
...
+        return self.lambd(img, target)
 
 
 class ConvertFromInts:
-    def __call__(self, image, boxes=None, labels=None):
-        return image.astype(np.float32), boxes, labels
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:81]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 9]
```diff
@@ -88,44 +88,48 @@ def __init__(self, mean, std, normalize_coef):
         self.std = np.array(std, dtype=np.float32)
         self.normalize_coef = normalize_coef
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:91]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 10]
```diff
@@ -88,44 +88,48 @@ def __init__(self, mean, std, normalize_coef):
...
         image /= self.normalize_coef
         image -= self.mean
         image /= self.std
-        return image.astype(np.float32), boxes, labels
+        return image.astype(np.float32), target
 
 
 class ToAbsoluteCoords:
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:100]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 11]
```diff
@@ -88,44 +88,48 @@ def __init__(self, mean, std, normalize_coef):
...
-
-        return image, boxes, labels
+        for i, x in enumerate(target):
+            x['bbox'] = boxes[i]
+        return image, target
 
 
 class ToPercentCoords:
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:113]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 12]
```diff
@@ -88,44 +88,48 @@ def __init__(self, mean, std, normalize_coef):
...
+            x['bbox'] = boxes[i]
+        return image, target
 
 
 class Resize:
     def __init__(self, size=300):
         self.size = size
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:129]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 13]
```diff
@@ -135,24 +139,24 @@ def __init__(self, lower=0.5, upper=1.5):
         assert self.upper >= self.lower, "contrast upper must be >= lower."
         assert self.lower >= 0, "contrast lower must be non-negative."
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:142]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 14]
```diff
@@ -135,24 +139,24 @@ def __init__(self, lower=0.5, upper=1.5):
...
+        return image, target
 
 
 class RandomHue:
     def __init__(self, delta=18.0):
         assert 0 <= delta <= 360.
         self.delta = delta
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:154]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 15]
```diff
@@ -161,28 +165,28 @@ def __init__(self):
                       (1, 0, 2), (1, 2, 0),
                       (2, 0, 1), (2, 1, 0))
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:168]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 16]
```diff
@@ -161,28 +165,28 @@ def __init__(self):
...
+        return image, target
 
 
 class ConvertColor:
     def __init__(self, current='BGR', transform='HSV'):
         self.transform = transform
         self.current = current
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:182]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 17]
```diff
@@ -206,21 +210,21 @@ def __init__(self, delta=32):
         assert delta <= 255.0
         self.delta = delta
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:213]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 18]
```diff
@@ -206,21 +210,21 @@ def __init__(self, delta=32):
...
             delta = random.uniform(-self.delta, self.delta)
             image += delta
-        return image, boxes, labels
+        return image, target
 
 
 class ToCV2Image:
-    def __call__(self, tensor, boxes=None, labels=None):
-        return tensor.cpu().numpy().astype(np.float32).transpose((1, 2, 0)), boxes, labels
+    def __call__(self, tensor, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:221]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 19]
```diff
@@ -206,21 +210,21 @@ def __init__(self, delta=32):
...
+        return tensor.cpu().numpy().astype(np.float32).transpose((1, 2, 0)), target
 
 
 class ToTensor:
-    def __call__(self, cvimage, boxes=None, labels=None):
-        return torch.from_numpy(cvimage.astype(np.float32)).permute(2, 0, 1), boxes, labels
+    def __call__(self, cvimage, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:226]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 20]
```diff
@@ -250,13 +254,15 @@ def __init__(self):
             (None, None),
         )
 
-    def __call__(self, image, boxes=None, labels=None):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:257]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 21]
```diff
@@ -324,17 +330,21 @@ def __call__(self, image, boxes=None, labels=None):
...
+                    x['label_idx'] = current_labels[i]
+                return current_image, target
 
 
 class Expand:
     def __init__(self, mean):
         self.mean = mean
 
-    def __call__(self, image, boxes, labels):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:344]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 22]
```diff
@@ -358,18 +368,26 @@ def __call__(self, image, boxes, labels):
...
+        for i, x in enumerate(target):
+            x['bbox'] = boxes[i]
 
-        return image, boxes, labels
+        return image, target
 
 
 class RandomMirror:
-    def __call__(self, image, boxes, classes):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:378]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 23]
```diff
@@ -411,19 +429,20 @@ def __init__(self):
         self.rand_brightness = RandomBrightness()
         self.rand_light_noise = RandomLightingNoise()
 
-    def __call__(self, image, boxes, labels):
+    def __call__(self, image, target):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:432]: Type hints

File: examples/torch/object_detection/utils/augmentations.py [hunk 24]
```diff
@@ -411,19 +429,20 @@ def __init__(self):
...
             distort = Compose(self.pd[1:])
-        im, boxes, labels = distort(im, boxes, labels)
-        return self.rand_light_noise(im, boxes, labels)
+        im, target = distort(im, target)
+        return self.rand_light_noise(im, target)
 
 
 class SSDAugmentation:
-    def __init__(self, size, mean, std, normalize_coef):
+    def __init__(self, size: int, mean: int, std: int, normalize_coef: int):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:445]: Type hints are invalid here

File: examples/torch/object_detection/utils/augmentations.py [hunk 25]
```diff
@@ -440,5 +459,9 @@ def __init__(self, size, mean, std, normalize_coef):
             Normalize(self.mean, self.std, self.normalize_coef)
         ])
 
-    def __call__(self, img, boxes, labels):
-        return self.augment(img, boxes, labels)
+    def __call__(self, img: Image, target: list):
```
Comment (vshampor) [inline examples/torch/object_detection/utils/augmentations.py:462]: Use `List`

**Review by vshampor — CHANGES_REQUESTED** [review]

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

**Comment by Meshford** [comment]
Jenkins please retry a build

File: examples/torch/object_detection/README.md (hunk 1 above)
Comment (Meshford) [reply examples/torch/object_detection/README.md:26]: done

File: examples/torch/object_detection/dataset.py (hunk 2 above)
Comment (Meshford) [reply examples/torch/object_detection/dataset.py]: done

File: examples/torch/object_detection/dataset.py (hunk 3 above)
Comment (Meshford) [reply examples/torch/object_detection/dataset.py]: done

File: examples/torch/object_detection/datasets/voc0712.py (hunk 5 above)
Comment (Meshford) [reply examples/torch/object_detection/datasets/voc0712.py]: done

File: examples/torch/object_detection/datasets/voc0712.py (hunk 4 above)
Comment (Meshford) [reply examples/torch/object_detection/datasets/voc0712.py]: done

File: examples/torch/object_detection/utils/augmentations.py (hunk 21 above)
Comment (Meshford) [reply examples/torch/object_detection/utils/augmentations.py]: done

File: examples/torch/object_detection/utils/augmentations.py (hunk 22 above)
Comment (Meshford) [reply examples/torch/object_detection/utils/augmentations.py]: done

File: examples/torch/object_detection/utils/augmentations.py (hunk 23 above)
Comment (Meshford) [reply examples/torch/object_detection/utils/augmentations.py]: done

File: examples/torch/object_detection/utils/augmentations.py (hunk 24 above)
Comment (Meshford) [reply examples/torch/object_detection/utils/augmentations.py]: done

File: examples/torch/object_detection/utils/augmentations.py (hunk 25 above)
Comment (Meshford) [reply examples/torch/object_detection/utils/augmentations.py]: done

File: examples/torch/object_detection/README.md [hunk 26]
```diff
@@ -37,13 +38,12 @@ This scenario demonstrates quantization with fine-tuning of SSD300 on VOC datase
   --disable-compression 
   ```
 - Run the following command to start compression with fine-tuning on GPUs:
-`python main.py -m train --config configs/ssd300_vgg_voc_int8.json --data <path_to_dataset> --log-dir=../../results/quantization/ssd300_int8`
-It may take a few epochs to get the baseline accuracy results.
+`python main.py -m train --config configs/ssd300_vgg_voc_int8.json --data <path_to_dataset> --log-dir=../../results/quantization/ssd300_int8 --weights=<path_to_checkpoint>`  
+ It may take a few epochs to get the baseline accuracy results.
+- Use `--weights` flag with the path to a compatible PyTorch checkpoint in order to load all matching weights from the checkpoint into the model - useful if you need to start compression-aware training from a previously trained uncompressed (FP32) checkpoint instead of performing compression-aware training from scratch. This flag optional, but highly recommended to use.
```
Comment (vshampor) [inline examples/torch/object_detection/README.md:43]: *is* optional

File: examples/torch/object_detection/README.md [hunk 27]
```diff
@@ -23,7 +23,8 @@ pip install -r examples/torch/requirements.txt
 This scenario demonstrates quantization with fine-tuning of SSD300 on VOC dataset.
 
 #### Dataset preparation
-- Download and extract VOC2007 and VOC2012 train/val and test data + devkit from [here](https://pjreddie.com/projects/pascal-voc-dataset-mirror/)
+- Download and extract in one `folder` train/val+test VOC2007 and train/val VOC2012 data from [here](https://pjreddie.com/projects/pascal-voc-dataset-mirror/)
```
Comment (vshampor) [inline examples/torch/object_detection/README.md:26]: Do not use backticks to highlight anything other than the entities on a file system. Use regular "folder" instead of `folder`. Highlight `<path_to_dataset>` with backticks instead.

File: examples/torch/object_detection/utils/augmentations.py (hunk 7 above)
Comment (vshampor) [reply examples/torch/object_detection/utils/augmentations.py:78]: You still have not fixed this.

File: examples/torch/object_detection/utils/augmentations.py (hunk 18 above)
Comment (vshampor) [reply examples/torch/object_detection/utils/augmentations.py:222]: Same

**Review by vshampor — CHANGES_REQUESTED** [review]

File: examples/torch/object_detection/README.md (hunk 26 above)
Comment (Meshford) [reply examples/torch/object_detection/README.md]: done

**Review by vshampor — APPROVED** [review]
